test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java

```diff
@@ -0,0 +1,102 @@
+package net.sourceforge.kolmafia.swingui;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.awt.Component;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import javax.swing.JRootPane;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.Document;
+import javax.swing.text.html.HTMLDocument;
+import javax.swing.text.html.HTMLWriter;
+import net.sourceforge.kolmafia.swingui.widget.RequestPane;
+import net.sourceforge.kolmafia.utilities.StringUtilities;
+import org.junit.jupiter.api.Test;
+
+class CalendarFrameTest {
+
+  private String writeDocumentToString(HTMLDocument htmlDocument) {
+    StringWriter sw = new StringWriter();
+    HTMLWriter w =
+        new HTMLWriter(sw, htmlDocument, 0, 10000) {
+          {
+            setLineLength(999_999);
+          }
+        };
+    try {
+      w.write();
+      return sw.toString();
+    } catch (IOException | BadLocationException e) {
+      e.printStackTrace();
+      return "";
```

Comment:
may as well `fail()` here, or even just let the error propagate to fail the test.

Reply:
OK